packages/hoppscotch-backend/src/shared-request/shared-request.service.ts

```diff
@@ -0,0 +1,259 @@
+   */
+  async createSharedRequest(
+    request: string,
+    properties: string | null = null,
+    userInfo: AuthUser,
+  ) {
+    const requestData = stringToJson(request);
+    if (E.isLeft(requestData))
+      return E.left(SHARED_REQUEST_INVALID_REQUEST_JSON);
+
+    const parsedProperties = stringToJson(properties);
+    if (E.isLeft(parsedProperties))
+      return E.left(SHARED_REQUEST_INVALID_PROPERTIES_JSON);
+
+    const generatedShortCode = await this.generateUniqueShortCodeID();
+    if (E.isLeft(generatedShortCode)) return E.left(generatedShortCode.left);
+
+    const createdSharedRequest = await this.prisma.shortcode.create({
+      data: {
+        id: generatedShortCode.right,
+        request: requestData.right,
+        properties: parsedProperties.right ?? undefined,
+        creatorUid: userInfo.uid,
+      },
+    });
+
+    this.pubsub.publish(
+      `shared_request/${createdSharedRequest.creatorUid}/created`,
+      this.cast(createdSharedRequest),
+    );
+
+    return E.right(this.cast(createdSharedRequest));
+  }
+
+  /**
+   * Fetch SharedRequest created by a User
+   *
+   * @param uid User Uid
+   * @param args Pagination arguments
+   * @returns Array of SharedRequest
+   */
+  async fetchUserSharedRequests(uid: string, args: PaginationArgs) {
+    const sharedRequests = await this.prisma.shortcode.findMany({
+      where: {
+        creatorUid: uid,
+      },
+      orderBy: {
+        createdOn: 'desc',
+      },
+      skip: args.cursor ? 1 : 0,
+      take: args.take,
+      cursor: args.cursor ? { id: args.cursor } : undefined,
+    });
+
+    const fetchedSharedRequests: SharedRequest[] = sharedRequests.map((code) =>
+      this.cast(code),
+    );
+
+    return fetchedSharedRequests;
+  }
+
+  /**
+   * Delete a SharedRequest
+   *
+   * @param sharedRequestID SharedRequest ID
+   * @param uid User Uid
+   * @returns Boolean on successful deletion
+   */
+  async revokeSharedRequest(sharedRequestID: string, uid: string) {
+    try {
+      const deletedSharedRequest = await this.prisma.shortcode.delete({
+        where: {
+          creator_uid_shortcode_unique: {
+            creatorUid: uid,
+            id: sharedRequestID,
+          },
+        },
+      });
+
+      this.pubsub.publish(
+        `shared_request/${deletedSharedRequest.creatorUid}/revoked`,
+        this.cast(deletedSharedRequest),
+      );
+
+      return E.right(true);
+    } catch (error) {
+      return E.left(SHARED_REQUEST_NOT_FOUND);
+    }
+  }
+
+  /**
+   * Update a created SharedRequest
+   * @param sharedRequestID SharedRequest ID
+   * @param uid User Uid
+   * @returns Updated SharedRequest
+   */
+  async updateSharedRequest(
+    sharedRequestID: string,
+    uid: string,
+    updatedProps: string,
+  ) {
+    if (!updatedProps) return E.left(SHARED_REQUEST_PROPERTIES_NOT_FOUND);
+
+    const parsedProperties = stringToJson(updatedProps);
+    if (E.isLeft(parsedProperties))
```

Comment:
There is a corner case for specific input for `properties` (null as string - see screenshot)

<img width="1273" alt="Screenshot 2023-11-02 at 11 58 50 AM" src="https://github.com/hoppscotch/hoppscotch/assets/30154643/47d97a34-bc54-41d7-9e78-48e02733640d">

```suggestion
    if (E.isLeft(parsedProperties) || !parsedProperties.right)
```